tests/tabs.ts

```diff
@@ -0,0 +1,387 @@
+        expect(validate_el(tabs, 1)).to.equal(true);
+        validate_prev(tabs, 0);
+        validate_RC(tabs, 4);
+
+        tabs.show(3);
+        expect(validate_el(tabs, 3)).to.equal(true);
+        validate_prev(tabs, 1);
+        validate_RC(tabs, 4);
+
+        tabs.show(0);
+        expect(validate_el(tabs, 0)).to.equal(true);
+        validate_prev(tabs, 3);
+
+        tabs.show(2);
+        expect(validate_el(tabs, 2)).to.equal(true);
+        validate_prev(tabs, 3);
+
+        let t4 = tabs.querySelector("#t4");
+        t4.click();
+        expect(validate_el(tabs, 3)).to.equal(true);
+
+        let t2 = tabs.querySelector("#t2");
+        create_slot_el(tabs);
+        t2.click();
+        expect(validate_el(tabs, 1)).to.equal(true);
+        validate_prev(tabs, 3);
+        validate_RC(tabs, 5);
+    });
+
+    it('row-column-accordion', async () => {
+        const el = await fixture (`
+            <div id='parent'>
+                <aalam-tabs fashion="xs:accordion;m:row;xl:column">
+                    <div slot="tab-title" id="t1">title1</div>
+                    <div slot="tab-body" id='b1'>body1</div>
+                    <div slot='tab-title' class='tab-active'
+                         id='t2'>title2</div>
+                    <div slot='tab-body' id='b2'>body2</div>
+                    <div slot="tab-title" class='tab-active'
+                         id="t3">title3</div>
+                    <div slot="tab-body" id='b3'>body3</div>
+                    <div slot='tab-title' class='tab-active'
+                         id='t4'>title4</div>
+                    <div slot='tab-body' id='b4'>body4</div>
+                </aalam-tabs>
+            </div>
+        `);
+        let tabs = el.querySelector("aalam-tabs");
+        expect(validate_el(tabs, 1)).to.equal(true);
+        expect(validate_el(tabs, 2)).to.equal(false);
+        expect(validate_el(tabs, 3)).to.equal(false);
+
+        await setViewport({width:630, height:640})
+        tabs.show(2);
+        expect(validate_el(tabs, 2)).to.equal(true);
+        validate_acc(tabs, 4);
+        expect(tabs._internal_fashion).to.equal('accordion');
+
+        await setViewport({width:1000, height:640});
+        create_slot_el(tabs);
+        window.dispatchEvent(new Event('resize'));
+        expect(tabs._internal_fashion).to.equal('row');
+        expect(validate_el(tabs, 2)).to.equal(true);
+
+        await setViewport({width:1640, height:680});
+        window.dispatchEvent(new Event('resize'));
+        validate_RC(tabs, 5);
+        expect(tabs._internal_fashion).to.equal('column');
+        expect(tabs._column_size.title).to.equal('30%');
+        expect(tabs._column_size.body).to.equal('70%');
+
+        let col = "title:50%;body:50%";
+        tabs.setAttribute("closize", col);
+        expect(tabs._column_size.title).to.equal('30%');
+        expect(tabs._column_size.body).to.equal('70%');
+        tabs.attributeChangedCallback('colsize', '', col);
```

Comment:
We dont need the attribute changed callback to be called, if you had setAttribute ("colsize", col) instead of "closize".